plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/internal/refresh/GMFHelper.java

```diff
@@ -130,40 +140,185 @@ private GMFHelper() {
      * 
      * @param node
      *            the GMF Node
+     * @param insetsAware
+     *            true to consider the draw2D figures insets. <strong>Warning:</strong> Those insets are based on the
+     *            current Sirius editParts and could become wrong if a developer customizes them.
+     * @param adaptBorderNodeLocation
+     *            Useful for specific border nodes, like in sequence diagrams, to center the border nodes if the
+     *            coordinate is 0 (x for EAST or WEST side, y for NORTH or SOUTH side)
      * 
      * @return the absolute location of the node relative to the origin (Diagram)
      */
-    public static Point getAbsoluteLocation(Node node) {
-        return getAbsoluteLocation(node, false);
+    public static Point getAbsoluteLocation(Node node, boolean insetsAware, boolean adaptBorderNodeLocation) {
```

Comment:
Method getAbsoluteLocation begins to "smell"...  It's the same for the GMFHelper class in general, but I don't know if it's easy/possible to refactor it

Reply:
You're probably right, but this is a first iteration (started 9 months ago), so I'd rather stop for now.